server/text_generation/utils.py

```diff
@@ -170,20 +172,62 @@ def initialize_torch_distributed():
     return torch.distributed.distributed_c10d._get_default_group(), rank, world_size
 
 
-def weight_hub_files(model_name, extension=".safetensors"):
+def weight_hub_files(model_name, revision=None, extension=".safetensors"):
     """Get the safetensors filenames on the hub"""
     api = HfApi()
-    info = api.model_info(model_name)
+    info = api.model_info(model_name, revision=revision)
     filenames = [s.rfilename for s in info.siblings if s.rfilename.endswith(extension)]
     return filenames
 
 
-def weight_files(model_name, extension=".safetensors"):
+def try_to_load_from_cache(model_name, revision, filename):
```

Comment:
Curious of the reason not to use `try_to_load_from_cache` from `huggingface_hub`?

Reply:
[It was bugged.](https://github.com/huggingface/huggingface_hub/pull/1316)